security/leaks.js

```diff
@@ -0,0 +1,140 @@
+window.addEventListener('load', async () => {
+    const sections = {};
+    ['added', 'removed', 'changed'].forEach((s) => {
+        sections[s] = document.getElementById(s);
+    });
+
+    function addBullet (html, section) {
+        const li = document.createElement('li');
+        li.innerHTML = html;
+        sections[section].appendChild(li);
+    }
+
+    function collectProps () {
+        const objCache = new Set();
+        // skip these
+        objCache.add(document);
+        objCache.add(window.self);
+        objCache.add(document.activeElement);
+        objCache.add(event?.currentTarget);
+        const recursionAllowed = new Set();
+        recursionAllowed.add('navigator');
+
+        function extractObjectProps (obj, bc = []) {
```

Comment:
bc? can you expand the name? I guess it's the object chain, right?

[edit] ah "breadcrumbs", let's use that